# ğŸª‘grayscale_ğŸ¬key_config
#	ãƒ‡ã‚£ãƒ¬ã‚¯ã‚¿ãƒ¼ï¼ˆDirectorï¼‰
#
#	ç´°ã‹ãªè¨­å®šã¯ä½•ã‚‚ã§ããªã„ã®ã§ã€ã‚½ãƒ¼ã‚¹ã‚’ç›´æ¥ã‚«ã‚¹ã‚¿ãƒã‚¤ã‚ºã™ã‚‹ã“ã¨
extends Node2D


# å€¤ã¯ãƒœã‚¿ãƒ³ç•ªå·ã€‚ãƒ¬ãƒãƒ¼ã¯ +1000
var key_config = {
	# ä»®æƒ³ã‚­ãƒ¼ï¼ˆï¼‘ï¼‰æ±ºå®šãƒœã‚¿ãƒ³ã€ãƒ¡ãƒƒã‚»ãƒ¼ã‚¸é€ã‚Šãƒœã‚¿ãƒ³
	&"VK_Ok" : -1,
	# ä»®æƒ³ã‚­ãƒ¼ï¼ˆï¼’ï¼‰ã‚­ãƒ£ãƒ³ã‚»ãƒ«ãƒœã‚¿ãƒ³ã€ãƒ¡ãƒ‹ãƒ¥ãƒ¼ãƒœã‚¿ãƒ³
	&"VK_Cancel" : -1,
	# ä»®æƒ³ã‚­ãƒ¼ï¼ˆï¼“ï¼‰ãƒ¡ãƒƒã‚»ãƒ¼ã‚¸æ—©é€ã‚Šãƒœã‚¿ãƒ³
	&"VK_FastForward" : -1,
}


# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ç”»é¢ã‚’å§‹ã‚ã‚‹ã‚¿ã‚¤ãƒŸãƒ³ã‚°ã§ä»¥ä¸‹ã‚’å‘¼å‡ºã™
func entry():
	$"KeyConfigArtist".entry()


func on_exit():
	print("ï¼»ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ï¼½ã€€å®Œäº†")
	# ï¼ˆæ³¨æ„ï¼‘ï¼ï¼”ï¼‰ã€€ä¾‹ãˆã°ã€ã‚·ãƒ¼ãƒ³ã®å¤–å´ã®ï¼‘éšå±¤ä¸Šã® `Director` ãƒãƒ¼ãƒ‰ã¸ã‚¢ã‚¯ã‚»ã‚¹ã™ã‚‹æ–¹æ³•
	$"../../Director".on_key_config_exited()


# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦
func on_process(delta):
	$"KeyConfigArtist".on_process(delta)


func on_unhandled_input(event):
	# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦
	$"KeyConfigArtist".on_unhandled_input(event)


# Called when the node enters the scene tree for the first time.
func _ready():
	pass
	# ï¼ˆæ³¨æ„ï¼’ï¼ï¼”ï¼‰ã€€ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ãŒè‡ªå‹•ã§å§‹ã¾ã‚‹ã®ãŒå›°ã‚‹ãªã‚‰ã€ä»¥ä¸‹ã‚’ã‚³ãƒ¡ãƒ³ãƒˆã‚¢ã‚¦ãƒˆã™ã‚‹ã“ã¨
	#self.entry()


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass
	# ï¼ˆæ³¨æ„ï¼“ï¼ï¼”ï¼‰ã€€ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ãŒè‡ªå‹•ã§å§‹ã¾ã‚‹ã®ãŒå›°ã‚‹ãªã‚‰ã€ä»¥ä¸‹ã‚’ã‚³ãƒ¡ãƒ³ãƒˆã‚¢ã‚¦ãƒˆã™ã‚‹ã“ã¨
	#self.on_process(delta)


func _unhandled_input(event):
	pass
	# ï¼ˆæ³¨æ„ï¼”ï¼ï¼”ï¼‰ã€€ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ãŒè‡ªå‹•ã§å§‹ã¾ã‚‹ã®ãŒå›°ã‚‹ãªã‚‰ã€ä»¥ä¸‹ã‚’ã‚³ãƒ¡ãƒ³ãƒˆã‚¢ã‚¦ãƒˆã™ã‚‹ã“ã¨
	#self.on_unhandled_input(event)


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ä»¥ä¸‹ã€æœ‰ã£ãŸã‚‰ä¾¿åˆ©ãªé–¢æ•°
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ãƒœã‚¿ãƒ³ç•ªå·ã€ã¾ãŸã¯ãƒ¬ãƒãƒ¼ç•ªå·ã‚’è¿”ã™ã€‚ãƒ¬ãƒãƒ¼ç•ªå·ã¯ +1000 ã—ã¦è¿”ã™ã€‚è©²å½“ãŒãªã‘ã‚Œã° -1 ã‚’è¿”ã™
func get_button_number_by_text(event_as_text):
	return $"KeyConfigArtist".get_button_number_by_text(event_as_text)


# âãƒœã‚¿ãƒ³ï¼‘â ã‚„ã€ âãƒ¬ãƒãƒ¼ï¼’â ã¨ã„ã£ãŸæ–‡å­—åˆ—ã‚’è¿”ã™ã€‚è©²å½“ãŒãªã‘ã‚Œã°ç©ºæ–‡å­—åˆ—ã‚’è¿”ã™
func get_button_name_by_number(button_number):
	return $"KeyConfigArtist".get_button_name_by_number(button_number)


# ãƒœã‚¿ãƒ³ç•ªå·ã‚’ã€ä»®æƒ³ã‚­ãƒ¼åã«å¤‰æ›ã€‚è©²å½“ãŒãªã‘ã‚Œã°ç©ºæ–‡å­—åˆ—
func get_virtual_key_name_by_button_number(button_number):
	return $"KeyConfigArtist".get_virtual_key_name_by_button_number(button_number)
